adapters/repos/db/lsmkv/segment_group_compaction.go

```diff
@@ -24,75 +24,78 @@ import (
 	"github.com/weaviate/weaviate/entities/cyclemanager"
 )
 
-func (sg *SegmentGroup) eligibleForCompaction() bool {
+func (sg *SegmentGroup) bestCompactionCandidatePair() []int {
 	sg.maintenanceLock.RLock()
 	defer sg.maintenanceLock.RUnlock()
 
-	// if true, the parent shard has indicated that it has
-	// entered an immutable state. During this time, the
-	// SegmentGroup should refrain from flushing until its
-	// shard indicates otherwise
-	if sg.isReadyOnly() {
-		return false
-	}
```

Comment:
Shouldn't we still check if the segment group is readonly before attempting compation?

Reply:
Thanks, that was an accidental delete